dspace-server-webapp/src/test/java/org/dspace/app/rest/authorization/GenericAuthorizationFeatureIT.java

```diff
@@ -0,0 +1,1709 @@
+                + "http://localhost/api/core/items/" + item1.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='"
+                + feature + "')]").doesNotExist());
+
+        // Verify community A admin doesn’t have this feature on item 2
+        getClient(communityAAdminToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item2.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='"
+                + feature + "')]").doesNotExist());
+
+
+        // grant item 1 admin REMOVE permissions on the item’s owning collection
+        // verify item 1 admin has this feature on item 1
+        context.turnOffAuthorisationSystem();
+        ResourcePolicy removePermission = ResourcePolicyBuilder.createResourcePolicy(context)
+            .withDspaceObject(collectionX)
+            .withAction(Constants.REMOVE)
+            .withUser(item1Writer)
+            .build();
+        context.restoreAuthSystemState();
+
+        // verify item 1 write has this feature on item 1
+        getClient(item1WriterToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item1.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='canMove')]")
+                .exists());
+
+        context.turnOffAuthorisationSystem();
+        ResourcePolicyBuilder.delete(removePermission.getID());
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void testCanMoveWriter() throws Exception {
+        testWriteUsersHavePermissionsItem("canMove", false);
+
+        // grant item 1 write REMOVE permissions on the item’s owning collection
+        context.turnOffAuthorisationSystem();
+        ResourcePolicy removePermission = ResourcePolicyBuilder.createResourcePolicy(context)
+            .withDspaceObject(collectionX)
+            .withAction(Constants.REMOVE)
+            .withUser(item1Writer)
+            .build();
+        context.restoreAuthSystemState();
+
+        String item1WriterToken = getAuthToken(item1Writer.getEmail(), password);
+        // verify item 1 write has this feature on item 1
+        getClient(item1WriterToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item1.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='canMove')]")
+                .exists());
+
+        context.turnOffAuthorisationSystem();
+        ResourcePolicyBuilder.delete(removePermission.getID());
+        context.restoreAuthSystemState();
```

Comment:
Same here, these lines shouldn't be needed.

Reply:
This has been adjusted